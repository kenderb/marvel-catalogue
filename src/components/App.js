import React from 'react';

import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import ComicDetail from '../containers/ComicDetail';
import NavBar from './NavBar';
import CharacterForm from '../containers/ComicForm';
import CharacterList from '../containers/ComicList';

const App = () => (
  <Router>
    <NavBar />
    <Switch>
      <Route path="/" exact>
        <CharacterForm />
        <CharacterList />
      </Route>
      <Route path="/comics/:id" component={ComicDetail} />
    </Switch>
  </Router>
);

export default App;
