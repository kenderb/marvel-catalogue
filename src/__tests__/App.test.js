import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { create } from 'react-test-renderer';
import { createStore, applyMiddleware } from 'redux';
import combineReducers from '../reducers';
import App from '../components/App';

test('App component renders correctly', () => {
  const store = createStore(combineReducers, applyMiddleware(thunk));
  const tree = create(
    <Provider store={store}>
      <App />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
