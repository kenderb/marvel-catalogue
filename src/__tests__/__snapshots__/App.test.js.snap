// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly 1`] = `
Array [
  <header>
    <nav>
      <ul>
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
             LOGO 
          </a>
        </li>
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
             List 
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "red",
      }
    }
    type="button"
  >
    <img
      alt="iron man"
      src="iron-man.png"
    />
  </button>,
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
    type="button"
  >
    <img
      alt="spider-man"
      src="spider-man.png"
    />
  </button>,
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
    type="button"
  >
    <img
      alt="thor"
      src="thor.png"
    />
  </button>,
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
    type="button"
  >
    <img
      alt="hulk"
      src="hulk.png"
    />
  </button>,
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
    type="button"
  >
    <img
      alt="wolverine"
      src="wolverine.png"
    />
  </button>,
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
    type="button"
  >
    <img
      alt="captain america"
      src="captain-america.png"
    />
  </button>,
  <select
    onChange={[Function]}
  >
    <option
      value="All"
    >
      All
    </option>
    <option
      value="Federico Blee"
    >
      Federico Blee
    </option>
    <option
      value="Sebastian Carrillo"
    >
      Sebastian Carrillo
    </option>
    <option
      value="John Tyler Christopher"
    >
      John Tyler Christopher
    </option>
    <option
      value="Christina Harrington"
    >
      Christina Harrington
    </option>
    <option
      value="Vc Cory Petit"
    >
      Vc Cory Petit
    </option>
    <option
      value="Mariko Tamaki"
    >
      Mariko Tamaki
    </option>
    <option
      value="Tom Brevoort"
    >
      Tom Brevoort
    </option>
    <option
      value="C Cafu"
    >
      C Cafu
    </option>
    <option
      value="Christopher Cantwell"
    >
      Christopher Cantwell
    </option>
    <option
      value="Vc Joe Caramagna"
    >
      Vc Joe Caramagna
    </option>
    <option
      value="Frank D'ARMATA"
    >
      Frank D'ARMATA
    </option>
    <option
      value="Alex Ross"
    >
      Alex Ross
    </option>
    <option
      value="Henry Abrams"
    >
      Henry Abrams
    </option>
    <option
      value="J.J Abrams"
    >
      J.J Abrams
    </option>
    <option
      value="Olivier Coipel"
    >
      Olivier Coipel
    </option>
    <option
      value="Nick Lowe"
    >
      Nick Lowe
    </option>
    <option
      value="Sara Pichelli"
    >
      Sara Pichelli
    </option>
    <option
      value="Dave Stewart"
    >
      Dave Stewart
    </option>
  </select>,
  <h1>
    Loading...
  </h1>,
]
`;

exports[`renders correctly 1`] = `
Array [
  <header>
    <nav>
      <ul>
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
             LOGO 
          </a>
        </li>
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
             List 
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "red",
      }
    }
    type="button"
  >
    <img
      alt="iron man"
      src="iron-man.png"
    />
  </button>,
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
    type="button"
  >
    <img
      alt="spider-man"
      src="spider-man.png"
    />
  </button>,
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
    type="button"
  >
    <img
      alt="thor"
      src="thor.png"
    />
  </button>,
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
    type="button"
  >
    <img
      alt="hulk"
      src="hulk.png"
    />
  </button>,
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
    type="button"
  >
    <img
      alt="wolverine"
      src="wolverine.png"
    />
  </button>,
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
    type="button"
  >
    <img
      alt="captain america"
      src="captain-america.png"
    />
  </button>,
  <select
    onChange={[Function]}
  >
    <option
      value="All"
    >
      All
    </option>
    <option
      value="Federico Blee"
    >
      Federico Blee
    </option>
    <option
      value="Sebastian Carrillo"
    >
      Sebastian Carrillo
    </option>
    <option
      value="John Tyler Christopher"
    >
      John Tyler Christopher
    </option>
    <option
      value="Christina Harrington"
    >
      Christina Harrington
    </option>
    <option
      value="Vc Cory Petit"
    >
      Vc Cory Petit
    </option>
    <option
      value="Mariko Tamaki"
    >
      Mariko Tamaki
    </option>
    <option
      value="Tom Brevoort"
    >
      Tom Brevoort
    </option>
    <option
      value="C Cafu"
    >
      C Cafu
    </option>
    <option
      value="Christopher Cantwell"
    >
      Christopher Cantwell
    </option>
    <option
      value="Vc Joe Caramagna"
    >
      Vc Joe Caramagna
    </option>
    <option
      value="Frank D'ARMATA"
    >
      Frank D'ARMATA
    </option>
    <option
      value="Alex Ross"
    >
      Alex Ross
    </option>
    <option
      value="Henry Abrams"
    >
      Henry Abrams
    </option>
    <option
      value="J.J Abrams"
    >
      J.J Abrams
    </option>
    <option
      value="Olivier Coipel"
    >
      Olivier Coipel
    </option>
    <option
      value="Nick Lowe"
    >
      Nick Lowe
    </option>
    <option
      value="Sara Pichelli"
    >
      Sara Pichelli
    </option>
    <option
      value="Dave Stewart"
    >
      Dave Stewart
    </option>
  </select>,
  <h1>
    Loading...
  </h1>,
]
`;
